### PR TITLE
initial commit for material chips

### PR DESCRIPTION
**Pull request by enigmij**
Signed-off-by: unknown <ritwij3000@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
+# Material_Chip
```
Comment (samuel-abhishek) [inline README.md:1]: Test report will be added as an artifact during the github release. Please remove it from the code repo.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+Add the following dependency in entry/package.json
+
+```
+"dependencies": {
+    "@ohos/materialchips": "file:./materialchips"
+  }
+```
+## Usage Instructions
+
+### -Input Chip
```
Comment (samuel-abhishek) [inline README.md:13]: Add a online description about each chip type.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+|                           |---- InputChip.ets  # Input Chip
+|     |---- README.MD  # installation and usage
+```
+## Code Contribution
+If you find any problems during usage, you can submit an [Issue](https://github.com/Applib-OpenHarmony/Material_UI_Chips
+/issues) to us. Of course, we also welcome you to send us [PR](https://github.com/Applib-OpenHarmony/Material_UI_FAB/pulls).
+
+## Open source License
+This project is based on [Apache License 2.0](https://github.com/Applib-OpenHarmony/Material_UI_Chips
+/blob/main/LICENSE), please enjoy and participate in open source freely.
```
Comment (samuel-abhishek) [inline README.md]: Extra space got added, therefore the URL is not linkified. Please check and handle.

File: materialchips/src/main/ets/components/MainPage/EditChip.ets [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+@Component
+export struct EditChip {
+  @State Dest: number= 0
+  @State Sub: number= 1
+  @State Tic: number= 0
+  @State Rad: number= 0
```
Comment (samuel-abhishek) [inline materialchips/src/main/ets/components/MainPage/EditChip.ets]: Use meaningful and understandable variable names.

**Review by samuel-abhishek — CHANGES_REQUESTED** [review]

File: materialchips/src/main/ets/components/MainPage/ActionChip.ets [hunk 5]
```diff
@@ -15,67 +15,188 @@
...
+      this.activetextcolor = '#000000'
+      this.chipcolor = '#DCDCDC'
+
+      this.activechipcolor = '#800080'
+    }
+  }
+
+  export class Model extends Properties {
+    private text: string | Resource = undefined
+    constructor(text: string | Resource) {
```
Comment (samuel-abhishek) [inline materialchips/src/main/ets/components/MainPage/ActionChip.ets]: Include the chiptext also in parent class. Child class not required in this scenario. Do these changes for other chip types as well.

File: materialchips/src/main/ets/components/MainPage/ChoiceChip.ets [hunk 6]
```diff
@@ -48,8 +48,8 @@ export struct ChipSet {
 
 @Component
 export struct ChoiceChip {
-  @State Dest: number= 1
-  @State Rad: number= 0
+  @State Delete: number= 1
```
Comment (samuel-abhishek) [inline materialchips/src/main/ets/components/MainPage/ChoiceChip.ets]: Please follow variable naming conventions.

File: materialchips/src/main/ets/components/MainPage/FilterChip.ets [hunk 7]
```diff
@@ -15,90 +15,179 @@
...
+      .backgroundColor(this.checked ? this.model.getColor() : this.model.getChipColor())
       .onClick(() => this.onClick())
-      .margin(this.Dest * 8)
+      .margin(this.Delete * 8)
       .clip(true)
     }
   }
+}
+export namespace FilterChip {
+
```
Comment (samuel-abhishek) [inline materialchips/src/main/ets/components/MainPage/FilterChip.ets:126]: Format the code by Ctrl+Alt+L

File: README.md [hunk 8]
```diff
@@ -40,6 +43,8 @@ struct Index {
 }
 ```
 ### -Choice Chip
+In sets that contain at least two options, choice chips represent a single selection.
```
Comment (samuel-abhishek) [inline README.md:45]: Please update the Readme as per the latest changes

**Review by samuel-abhishek — CHANGES_REQUESTED** [review]

File: materialchips/package.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@ohos/materialchips",
```
Comment (archanaatr) [inline materialchips/package.json:2]: follow the library naming rules shared... Update it to @ohos/material-chips

File: package.json [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    "directoryLevel": "project"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {
+    "hypium": "^1.0.0",
+    "@ohos/hvigor": "1.0.6",
+    "@ohos/hvigor-ohos-plugin": "1.0.6",
+    "@ohos/materialchips": "file:./materialchips"
```
Comment (archanaatr) [inline package.json]: remove the chip lib dependency from the main file

File: entry/package.json [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  "ohos": {
+    "org": "huawei",
+    "buildTool": "hvigor",
+    "directoryLevel": "module"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {
+    "@ohos/materialchips": "file:./materialchips"
```
Comment (archanaatr) [inline entry/package.json]: change the lib name to material-chips

File: entry/src/main/ets/MainAbility/app.ets [hunk 12]
```diff
@@ -0,0 +1,8 @@
+export default {
```
Comment (archanaatr) [inline entry/src/main/ets/MainAbility/app.ets:16]: Add copyright in all the ets files

File: materialchips/src/main/ets/components/MainPage/InputChip.ets [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+@Component
+export struct InputChip {
+  @State delete: number= 1
```
Comment (archanaatr) [inline materialchips/src/main/ets/components/MainPage/InputChip.ets]: format all the etc files . Open each file and perform Ctrl + Alt + L

File: materialchips/src/main/ets/components/MainPage/InputChip.ets [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+  @State tickSize: number= 0
+  @State radius: number= 0
+  @State checked: boolean = false
+  @State private model: InputChip.Model = undefined
+
+  build() {
+    Column() {
+      Row() {
+        Image(this.model.getImage())
```
Comment (archanaatr) [inline materialchips/src/main/ets/components/MainPage/InputChip.ets]: Add validation of null and undefined values on model. Recheck and add for all such case

File: materialchips/src/main/ets/components/MainPage/InputChip.ets [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+    }
+  }
+}
+
+export namespace InputChip {
+
+  export class Model {
+    private text: string | Resource = ""
+    private img: string | Resource = ""
+    private textSize: number= 20
```
Comment (archanaatr) [inline materialchips/src/main/ets/components/MainPage/InputChip.ets]: Mark all dimentions as vp and text sizes as fp

File: materialchips/src/main/ets/components/MainPage/EditChip.ets [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+@Component
+export struct EditChip {
+  @State Delete: number= 0
```
Comment (archanaatr) [inline materialchips/src/main/ets/components/MainPage/EditChip.ets]: variable naming convention to be followed

File: materialchips/src/main/ets/components/MainPage/EditChip.ets [hunk 17]
```diff
@@ -0,0 +1,174 @@
...
+            Circle({ width: '1vp', height: '1vp' })
+              .fill(this.model.getChipColor())
+              .borderRadius('8vp')
+              .scale({ x: this.Radius, y: this.Radius })
+              .clip(true)
+            Text(this.text)
+              .fontSize(this.Delete * this.model.getTextSize())
+              .fontColor(this.model.getTextColor())
+              .margin({
+                left: this.Delete * 12,
```
Comment (archanaatr) [inline materialchips/src/main/ets/components/MainPage/EditChip.ets]: check to add as vp

**Review by archanaatr — COMMENTED** [review]
please check the comments

File: README.md [hunk 18]
```diff
@@ -93,9 +93,9 @@ struct Index {
   private filterchip2: FilterChip.Model = new FilterChip.Model()
   private filterchip3: FilterChip.Model = new FilterChip.Model()
   aboutToAppear() {
-    this.filterchip1.setText("Op1").setTextSize(20)
-    this.filterchip2.setText("Op2").setTextSize(20)
-    this.filterchip3.setText("Op3").setTextSize(20)
+    this.filterchip1.setText("Op1").setTextSize('20vp')
```
Comment (samuel-abhishek) [inline README.md]: use "fp" for textSize/fontSize and for other use "vp". Check this in all the files

File: entry/package.json [hunk 19]
```diff
@@ -9,7 +9,5 @@
   "description": "example description",
   "repository": {},
   "license": "ISC",
-  "dependencies": {
-    "@ohos/materialchips": "file:./materialchips"
-  }
+  "dependencies": {}
```
Comment (samuel-abhishek) [inline entry/package.json]: The lib dependency must be present here i.e entry's package.json. Remove the dependency from project's package.json

**Pull request merged**